### PR TITLE
[AlecAivazis/survey#96] add support for tab completion on input

### PR DESCRIPTION
**Pull request by lucassabreu**
related to AlecAivazis/survey#96

### CODE REVIEW

**Comment by lucassabreu** [comment]
i will create a new section similar to "filtering options" on how to use the new feature. but the code i think is done

File: input.go [hunk 1]
```diff
@@ -31,11 +44,90 @@ var InputQuestionTemplate = `
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}
 {{- if .ShowAnswer}}
   {{- color "cyan"}}{{.Answer}}{{color "reset"}}{{"\n"}}
+{{- else if .PageEntries -}}
+  {{- .Answer}} [Use arrows to navegate, enter to select, type to complement answer]
```
[the remark this answers is not in the document]
Comment (lucassabreu) [reply input.go]: i changed "navigate" to "move", to be like the select and multiselect... continue is better the "complement answer"... thank you

[the remark this answers is not in the document]
Comment (infinitepr0) [reply input.go]: Consider marking them resolved once done!

[the remark this answers is not in the document]
Comment (lucassabreu) [reply input.go]: done

File: input.go [hunk 2]
```diff
@@ -31,11 +44,90 @@ var InputQuestionTemplate = `
...
+  {{- .Answer}} [Use arrows to move, enter to select, type to continue]
+  {{- "\n"}}
+  {{- range $ix, $choice := .PageEntries}}
+    {{- if eq $ix $.SelectedIndex }}{{color $.Config.Icons.SelectFocus.Format }}{{ $.Config.Icons.SelectFocus.Text }} {{else}}{{color "default"}}  {{end}}
+    {{- $choice.Value}}
+    {{- color "reset"}}{{"\n"}}
+  {{- end}}
 {{- else }}
   {{- if and .Help (not .ShowHelp)}}{{color "cyan"}}[{{ print .Config.HelpInput }} for help]{{color "reset"}} {{end}}
+  {{- if and .Suggest }}{{color "cyan"}}[{{ print .Config.SuggestInput }} for suggestions]{{color "reset"}} {{end}}
```
Comment (AlecAivazis) [inline input.go:57]: this prints `[? for help][tab for suggestions]` whereas the other inputs have multiple instructions between the same []. Mind changing this so it renders like `[? for help, tab for suggestions]`?

**Review by AlecAivazis — CHANGES_REQUESTED** [review]

File: input.go (hunk 2 above)
Comment (lucassabreu) [reply input.go]: change made

**Review by AlecAivazis — CHANGES_REQUESTED** [review]
Sorry it took so long for me to pull this down and take a serious look at it. Life was insanely busy but I found some time.

So in general, I LOVE this feature. On top of the killer feature, you took all of the time to add all of the different bits to get this through - awesome!

I've run it on cmd.exe and the ubuntu subsystem on windows and it runs like a charm. One minor request - can you make tab cycle to the next item (as well as down arrow)? This reflects the way that screen readers interact with consoles so we can be more accesible.

**Comment by lucassabreu** [comment]
thank you @AlecAivazis, i am glad you liked it.

i implemented the tab to go to next item for the suggestions now.

i also realized the select and multiselect does not have this behavior and will implement on then.

**Review by AlecAivazis — APPROVED** [review]
Awesome, thanks for adding that! I think this is finally good to go. Sorry again it took so long!

**Comment by AlecAivazis** [comment]
@lucassabreu this PR seems to have broken the ability for the user to use arrow keys in order to navigate the input string :(

**Comment by lucassabreu** [comment]
hi @AlecAivazis , I can look into it, but can you explain how to reproduce it? is just about moving the cursor on the input?

**Comment by AlecAivazis** [comment]
awesome! If you run any of the tests in `input.go` you'll see what i mean if you type a few characters and then try to press an arrow key to move left or right.

**Comment by AlecAivazis** [comment]
The issue is because input [calls RuneReader.ReadRune](https://github.com/AlecAivazis/survey/blob/master/input.go#L160) instead of [ReadLine](https://github.com/AlecAivazis/survey/blob/master/terminal/runereader.go#L34) which handles new characters inserted at arbitrary places in the input.

**Comment by AlecAivazis** [comment]
I tried to sneak in some time this morning to get started on this but have to get back to work and might not have time to see it through to the end for a little. [Here](https://github.com/AlecAivazis/survey/tree/input-readline-with-suggestions) is my branch if you want to take a look. The input and help tests work but I haven't tried adding any suggestions yet. We probably want to add an example with suggestions to the test suite so we can ensure the functionality stays.

**Comment by lucassabreu** [comment]
right i will try to continue from the branch. i think i left some tests... but will try to add more if needed.

**Comment by AlecAivazis** [comment]
Thanks for picking this up! I really appreciate it

**Comment by lucassabreu** [comment]
trying to fix in the pr #361

**Pull request merged**